### ISSUE TITLE
onnx使用问题

### ISSUE THREAD

**Issue by mxlive** [Picsart-AI-Research/MI-GAN#22]
新手求救，有没有大佬指导下onnx模型，为什么我按照参数传值，输出的图，都是错误的
![10mask](https://github.com/user-attachments/assets/b0ca3d3e-b3ff-41a1-a95b-b2620511445b)
![10image](https://github.com/user-attachments/assets/cdf6059c-c392-43dd-a085-cbdfa0bc2a65)
![output_image](https://github.com/user-attachments/assets/6937f24d-0419-4240-a8b4-77310c0fff95)
代码如下：
import onnxruntime as ort
import numpy as np
import cv2
session = ort.InferenceSession("inpainting/model/migan_pipeline_v2.onnx")
image = cv2.imread('10image.png') 
mask = cv2.imread('10mask.png', cv2.IMREAD_GRAYSCALE)
image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
image = np.transpose(image, (2, 0, 1))
image = np.expand_dims(image, axis=0)
mask = np.expand_dims(mask, axis=0)
mask = np.expand_dims(mask, axis=1)
image = image.astype(np.uint8)
mask = mask.astype(np.uint8)
inputs = {
    'image': image,
    'mask': mask
}
outputs = session.run(None, inputs)
result = outputs[0]
result = result.squeeze()
result = np.transpose(result, (1, 2, 0))
result = cv2.cvtColor(result, cv2.COLOR_RGB2BGR)
cv2.imshow('Result', result)
cv2.waitKey(0)
cv2.destroyAllWindows()

**Comment by xueewang** [comment]
You could try to invert the masks.

Here is my code that you can try (refer to [create_onnx_pipeline](https://github.com/Picsart-AI-Research/MI-GAN/blob/2381ef9d322caa4f90550f4b7072a6f681efb8c2/scripts/create_onnx_pipeline.py#L333C1-L343C53)):

```
import onnxruntime as ort
import numpy as np
from PIL import Image
from scripts.demo import read_mask
import torch

img_path = './examples/places2_512_object/images/20.png'
mask_path = './examples/places2_512_object/masks/20.png'

img = Image.open(img_path)
mask = read_mask(mask_path, invert=True)

ort_sess = ort.InferenceSession('models/migan_pipeline_v2.onnx')

with torch.no_grad():
    img = np.expand_dims(img, 0).transpose(0, 3, 1, 2)
    mask = np.expand_dims(mask, (0, 1))
    result_image = ort_sess.run(None, {'image': img, 'mask':  mask})[0]

result_image = result_image[0].transpose(1, 2, 0)
result_image = Image.fromarray(result_image)
result_image.show()
```

**Comment by mxlive** [comment]
> You could try to invert the masks.
> 
> Here is my code that you can try (refer to [create_onnx_pipeline](https://github.com/Picsart-AI-Research/MI-GAN/blob/2381ef9d322caa4f90550f4b7072a6f681efb8c2/scripts/create_onnx_pipeline.py#L333C1-L343C53)):
> 
> ```
> import onnxruntime as ort
> import numpy as np
> from PIL import Image
> from scripts.demo import read_mask
> import torch
> 
> img_path = './examples/places2_512_object/images/20.png'
> mask_path = './examples/places2_512_object/masks/20.png'
> 
> img = Image.open(img_path)
> mask = read_mask(mask_path, invert=True)
> 
> ort_sess = ort.InferenceSession('models/migan_pipeline_v2.onnx')
> 
> with torch.no_grad():
>     img = np.expand_dims(img, 0).transpose(0, 3, 1, 2)
>     mask = np.expand_dims(mask, (0, 1))
>     result_image = ort_sess.run(None, {'image': img, 'mask':  mask})[0]
> 
> result_image = result_image[0].transpose(1, 2, 0)
> result_image = Image.fromarray(result_image)
> result_image.show()
> ```

谢谢，是对的，昨天我尝试了下，翻转掩码
